### ISSUE TITLE
mapPlot() problem with 'aeqd' and 'aitoff' projections

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1320]
I rebuilt the projection test suite that is posted at http://dankelley.github.io/r/2015/04/03/oce-proj.html and a lot of the old problems with spurious lines are now gone. There are still problems with some of the projections (and they are listed) but this posting is to illustrate one specifically, so I can work on code specifically for this one. The snapshot illustrates.

<img width="658" alt="screen shot 2017-10-30 at 7 33 28 pm" src="https://user-images.githubusercontent.com/99469/32198953-4d42aade-bda9-11e7-9c1b-1b62756ed4b3.png">

**Comment by dankelley** [comment]
I've started a new issue repo at https://github.com/dankelley/oce-issues/tree/master/13xx/1320 and (as of now) the test code and a result are as follows. The problem is quite evident in the top-left corner.
```R
library(oce)
data(coastlineWorld)
p <- "+proj=aitoff +lon_0=-45"
if (!interactive()) png("1320a.png")
par(mar=c(2, 2, 1, 1))
par(xpd=NA)
mapPlot(coastlineWorld, projection=p, col="lightgray",
        longitudelim=c(-80,0), latitudelim=c(0,79))
if (!interactive()) dev.off()
```

![1320a](https://user-images.githubusercontent.com/99469/32739378-019bd8e4-c876-11e7-8a12-a26b86b7dce3.png)

**Comment by dankelley** [comment]
This is fixed now, for both projections, in the 70eef7cd6780a9f73b362194a694674cfa3544fb commit of the "develop" branch. 

The test repo at 70eef7cd6780a9f73b362194a694674cfa3544fb tests both projections. In the graphs it makes, I trace a coastline portion that I was using as a test case. I am reasonably confident in the fix, because it stemmed from some sloppy C coding that is now written in a much simpler manner and so fairly easy to validate by eye.

I rebuilt my blog item on projections (http://dankelley.github.io/r/2015/04/03/oce-proj.html) and most of those test cases look good now. Still, I consider `coastlineCut` to be quite a flakey beast (it's not going to work well if a pole is in view) and so I certainly don't consider projections to be a "done" issue, despite that fact that I am closing this particular instance.
